File: src/Overmoney.Api/DataAccess/Transactions/Transaction.cs
namespace Overmoney.Api.DataAccess.Transactions;

public class Transaction
{
    public long Id { get; init; }
    public int AccountId { get; init; }
    public int UserId { get; init; }
    public int PayeeId {  get; init; }
    public int CategoryId { get; init; }
    public DateTime TransactionDate {  get; init; }
    public TransactionType TransactionType { get; init; }
    public string? Note { get; init; } 
}

public enum TransactionType
{
    Outcome = 0,
    Income = 1,
    Transfer = 2
}

File: src/Overmoney.Portal/src/data_access/models/transaction.ts
import type { Attachment } from "./attachment";
import type { Category } from "./category";
import type { Payee } from "./payee";
import type { Wallet } from "./wallet";

export type Transaction = {
    id: number;
    userId: number;
    wallet: Wallet;
    payee: Payee;
    category: Category;
    transactionDate: Date;
    transactionType: TransactionType;
    note?: string;
    amount: number;
    attachments?: Array<Attachment>;
}

export enum TransactionType {
    Outcome = 0,
    Income = 1,
    Transfer = 2
}